src/main/java/com/softserve/messaging/service/KafkaConsumer.java

```diff
@@ -0,0 +1,40 @@
+package com.softserve.messaging.service;
+
+import com.softserve.messaging.dto.MailTemplateDto;
+import com.softserve.messaging.model.MessageHandler;
+import com.softserve.messaging.repository.MessageHandlerRepo;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.springframework.kafka.annotation.KafkaListener;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class KafkaConsumer {
+
+    private final MailService mailService;
+    private final MessageHandlerRepo messageHandlerRepo;
+    private final MailTemplateDirector mailTemplateDirector;
+
+    @KafkaListener(topics = "${kafka.topic.email-receipt}", containerFactory = "kafkaListenerContainerFactory")
```

Comment:
Nit: <related to the comment on `topic` property>
just pass the name of the topic directly

Reply:
you don't need containerFactory

Reply:
done